data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/ShutdownHandler.java

```diff
@@ -0,0 +1,34 @@
+package com.amazon.dataprepper.pipeline.server;
+
+import com.amazon.dataprepper.DataPrepper;
+import java.io.IOException;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HttpHandler to handle requests to shut down the data prepper instance
+ */
+public class ShutdownHandler implements HttpHandler {
+
+    private final DataPrepper dataPrepper;
+    private final static Logger LOG = LoggerFactory.getLogger(ShutdownHandler.class);
+
+    public ShutdownHandler(final DataPrepper dataPrepper) {
+        this.dataPrepper = dataPrepper;
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            dataPrepper.shutdown();
```

Comment:
Just curious - if dataPrepper is already shutdown, will we return a 200 or 500?

Reply:
So once dataPrepper is fully shut down, then the server will shut down as well.